### ISSUE TITLE
The tests for twisted.python.filepath shouldn't use eval()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5908]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5908|
|Type|enhancement|
|Created|2012-08-23 16:16:30Z|

Constructing Unix permission octets with `eval()` is silly; we should fix `twisted.test.test_paths` not to do so.

Attachments:

* [patch5908.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/515/515920415c3d73bbc5fcaa2dfd4168e4cddce8a9/9e6a31eb2f6bf8611a759efd63cc1a7b953f050c.patch) (1062 bytes) - added by Petit_Dejeuner on 2012-10-29 02:18:58Z - Replaces call to 'eval()' with call to 'int(var, 8)'.

<details><summary>Searchable metadata</summary>

```
trac-id__5908 5908
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1345738590000000 1345738590000000
changetime__1352995998000000 1352995998000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="dinabina's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|dinabina set owner to dinabina<br>dinabina set status to `assigned`|
|-|-|

Is this a simple matter of changing the use of eval(octet_string) to int(octet_string, 8), where octet_string is built using the existing for-loop.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst removed owner<br>@itamarst set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @therve|
|-|-|

Looks good, applying.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [36408]) Replace usage of eval by int in twisted.tests.test_path

Author: Petit_Dejeuner
Reviewer: therve
Fixes: [#5908](https://github.com/twisted/twisted/issues/5908)
